src/utils/getTypeString.ts

```diff
@@ -112,3 +119,52 @@ export function generateLiteralTypeString(value: string): string {
   }
   return `int_const ${value}`;
 }
+
+export function generateExpressionTypeString(type: TypeNode): string {
+  if (type instanceof PointerType) {
+    if (type.to instanceof MappingType) return generateExpressionTypeString(type.to);
+    else
+      return `${generateExpressionTypeString(type.to)} ${type.location}${
+        type.kind !== undefined ? ' ' + type.kind : ''
+      }`;
+  }
+  if (type instanceof FunctionType) {
+    const mapper = (node: TypeNode) => generateExpressionTypeString(node);
+
+    const argStr = type.parameters.map(mapper).join(',');
+
+    let retStr = type.returns.map(mapper).join(',');
+
+    retStr = retStr !== '' ? ` returns (${retStr})` : retStr;
+
+    const visStr = type.visibility !== FunctionVisibility.Internal ? ` ` + type.visibility : '';
+    const mutStr = type.mutability !== 'nonpayable' ? ' ' + type.mutability : '';
+
+    return `function ${
+      type.name !== undefined ? type.name : ''
+    }(${argStr})${mutStr}${visStr}${retStr}`;
+  }
+  if (type instanceof UserDefinedType) {
+    if (!(type.definition instanceof UserDefinedValueTypeDefinition)) return type.pp();
+    return type.definition.underlyingType.typeString;
+  }
+  if (type instanceof ArrayType) {
+    return `${generateExpressionTypeString(type.elementT)}[${
+      type.size !== undefined ? type.size : ''
+    }]`;
+  }
+  if (type instanceof MappingType) {
+    return `mapping(${generateExpressionTypeString(type.keyType)} => ${generateExpressionTypeString(
+      type.valueType,
+    )})`;
+  }
+  if (type instanceof TupleType) {
+    return `tuple(${type.elements
+      .map((element) => generateExpressionTypeString(element))
+      .join(',')})`;
+  }
+  if (type instanceof TypeNameType) {
+    return `type(${generateExpressionTypeString(type.type)})`;
+  }
+  return type.pp();
```

Comment:
Instead of returning `type.pp()` if execution gets to this point, this should be an if that checks if `type` is an instance of any of the TypeNode subclasses for which we can safely use the `pp` function. Then an error should be thrown if the node is not any of those types. That way, if solc-typed-ast is updated ands adds a new TypeNode it'll produce an error rather than just using the `pp` function